runtime/doc/lua.txt

```diff
@@ -3296,6 +3296,23 @@ filter({f}, {src}, {...})                                  *vim.iter.filter()*
     See also: ~
       • |Iter:filter()|
 
+flatten({src}, {depth}, {...})                            *vim.iter.flatten()*
```

Comment:
`flat` is shorter and is what https://github.com/neovim/neovim/issues/26454 specifies.
```suggestion
flat({src}, {depth}, {...})                            *vim.iter.flat()*
```